### ISSUE TITLE
[BUG] Problem with potentialAction compilation

### ISSUE THREAD

**Issue by lulzmachine** [prometheus-msteams/prometheus-msteams#110]
Thanks for the prometheus-msteams-project! It has really been clutch for us to get the alerts system going

**Describe the bug**
We had a template that ended in something like this:
```
  ]
  {{ if .CommonAnnotations.runbook_url }}
  ,"potentialAction": [
  {
        "@context": "http://schema.org",
        "@type": "ViewAction",
        "name": "Runbook",
        "target": [
            "{{ reReplaceAll "_" "\\\\_" .CommonAnnotations.runbook_url }}"
        ]
    }
  ]
  {{- end -}}
```

But we ran into problems trying to use it:

got error: invalid character '\n' in string literal, wantErr false

This seems to be due to the function "concatKeyValue" on line 124 of templated_card.go requiring a JSON object to start with "[{" in order to get the required escaping

**Version**
Latest

**To Reproduce**
In templated_card_test.go, change out "templateFile" to something similar to above

**Expected behavior**
Expected valid JSON to go through the template compiler. I don't have too much insight into why the code is like it is, but my feeling is that template_compiler does a bit too much. Don't you just want to run a go template compile and then load in the result as json?

**Additional context**
Quickfix is to make sure "[{" is just next to each other, but this did take quite a few hours for us to solve. Hence the bug report

**Comment by bzon** [comment]
Thanks for submitting this issue.

> I don't have too much insight into why the code is like it is, but my feeling is that template_compiler does a bit too much.

It has been a very long time since that code was written and I think it is time that we refactor the templating code or a new implementation for the `card.Converter` interface.

> Quickfix is to make sure "[{" is just next to each other, but this did take quite a few hours for us to solve. Hence the bug report

Is this fix in code or in your template file?

**Comment by lulzmachine** [comment]
@bzon Thanks for the quick response. I meant that we changed our template to have "[{" instead of "[\n\t{" as a workaround

**Comment by Knappek** [comment]
thank @lulzmachine for filing this issue. I've been thinking about rewriting this part for quite some time, I'll probably start soon with it as we are observing similar kind of issues again and again.

**Comment by Knappek** [comment]
>I don't have too much insight into why the code is like it is, but my feeling is that template_compiler does a bit too much.
Just fyi, The tricky part in the code is that we need to split an alert that is too big into multiple chunks as MS teams has a byte size limit. 

But as said before, I will now start and try to make the code easier (it was my first experiences with Go at that time :D)

**Comment by bzon** [comment]
This is now fixed from #123
